src/mgmt/tests/mocks.c

```diff
@@ -274,31 +274,40 @@ void mock_ds_mgmt_pool_list_cont_teardown(void)
 	}
 }
 
-int			ds_mgmt_pool_query_return;
-uuid_t			ds_mgmt_pool_query_uuid;
-daos_pool_info_t	ds_mgmt_pool_query_info_out;
-daos_pool_info_t	ds_mgmt_pool_query_info_in;
-void			*ds_mgmt_pool_query_info_ptr;
-d_rank_list_t		*ds_mgmt_pool_query_ranks_out;
+int              ds_mgmt_pool_query_return;
+uuid_t           ds_mgmt_pool_query_uuid;
+daos_pool_info_t ds_mgmt_pool_query_info_out;
+daos_pool_info_t ds_mgmt_pool_query_info_in;
+void            *ds_mgmt_pool_query_info_ptr;
+d_rank_list_t   *ds_mgmt_pool_query_enabled_ranks_out;
+d_rank_list_t   *ds_mgmt_pool_query_disabled_ranks_out;
 
 int
-ds_mgmt_pool_query(uuid_t pool_uuid, d_rank_list_t *svc_ranks, d_rank_list_t **ranks,
-		   daos_pool_info_t *pool_info, uint32_t *pool_layout_ver,
-		   uint32_t *upgrade_layout_ver)
+ds_mgmt_pool_query(uuid_t pool_uuid, d_rank_list_t *svc_ranks, d_rank_list_t **enabled_ranks,
+		   d_rank_list_t **disabled_ranks, daos_pool_info_t *pool_info,
+		   uint32_t *pool_layout_ver, uint32_t *upgrade_layout_ver)
 {
 	/* If function is to return with an error, pool_info and ranks will not be filled. */
 	if (ds_mgmt_pool_query_return != 0)
 		return ds_mgmt_pool_query_return;
 
 	uuid_copy(ds_mgmt_pool_query_uuid, pool_uuid);
 	ds_mgmt_pool_query_info_ptr = (void *)pool_info;
+	if (pool_info != NULL && (pool_info->pi_bits & DPI_ENGINES_ENABLED) != 0) {
```

Comment:
nit: might look nicer to perform the pool_info NULL check once and indent the rest of the checks

Reply:
- [ ] Improve code readability